docs/guides/typescript.md

```diff
@@ -253,4 +188,58 @@ type State =
 /* ... */
 ```
 
-:::
+## Config Objects
+
+Machine config objects can be typed. This is useful when defining a machine config object _outside_ of the `createMachine(...)` function, and helps prevent [inference errors](https://github.com/davidkpiano/xstate/issues/310). The first and third generic parameters for `MachineConfig<TContext, TSchema, TEvent>` are the same as the first and second parameters for `createMachine<TContext, TEvent, TTypeState>` respectively. The second generic parameter 'TSchema' accepts an object type with a single key 'states' which lists the possible states of machines accepting the configuration:
+
+```ts
+import { MachineConfig } from 'xstate';
+
+type myStateSchema = {
+  states: {
+    stopped: {};
+    started: {};
+  };
+};
+
+const myMachineConfig: MachineConfig<TContext, myStateSchema, TEvent> = {
+  id: 'controller',
+  initial: 'stopped',
+  states: {
+    stopped: {
+      /* ... */
+    },
+    started: {
+      /* ... */
+    }
+  }
+  // ...
+};
+```
+
+## Actions
+
+The `send` action on the interpreted machine `interpret(stateMachine)` isn't always type safe. To get typechecking for this function signature, use the following pattern:
+
+```ts
+type UserEvents = {
+  type: 'TEST';
+  value: string;
+};
+
+const service = interpret(stateMachine);
+
+// This will compile
+service.send({ type: 'TEST', value: 'testvalue' });
+
+// This will have a compile error on the `value` type
+service.send({ type: 'TEST', value: 1 });
+```
+
+If you use the following pattern, you'll lose type safety, so both of these will compile:
```

Comment:
Not sure if this is accurate anymore (if it is, it will soon not be)

Reply:
In #2235 I verified that it's currently true.